services/movie_session.py

```diff
@@ -0,0 +1,55 @@
+import datetime
+
+from db.models import models
+from db.models import MovieSession, Movie, CinemaHall
+
+
+def create_movie_session(
+        movie_show_time: datetime.datetime,
+        movie_id: int,
+        cinema_hall_id: int
+) -> None:
+    movie_ = Movie.objects.get(id=movie_id)
+    hall = CinemaHall.objects.get(id=cinema_hall_id)
+    MovieSession.objects.create(
+        show_time=movie_show_time,
+        movie=movie_,
+        cinema_hall=hall
+    )
+
+
+def get_movies_sessions(
+        session_date: str = None
+) -> models.QuerySet:
+    queryset = MovieSession.objects.all()
+    if session_date:
+        queryset = queryset.filter(show_time__date=session_date)
+    return queryset
+
+
+def get_movie_session_by_id(movie_session_id: int) -> models.QuerySet:
+    return MovieSession.objects.get(id=movie_session_id)
+
+
+def update_movie_session(
+        session_id: int,
+        show_time: datetime.datetime = None,
+        movie_id: int = None,
+        cinema_hall_id: int = None

```

Comment:
use typehint like: Optional[YOUR_VALUE_TYPE] = None
for example: list that contains string elements and can be None has typehint Optional[list[str]]

Reply:
not fixed